modules/integration/tests-integration/tests-service/src/test/java/org/wso2/carbon/ml/model/test/LearningAlgorithmsTestCase.java

```diff
@@ -0,0 +1,328 @@
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.testng.SkipException;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.ml.integration.common.utils.MLBaseTest;
+import org.wso2.carbon.ml.integration.common.utils.MLHttpClient;
+import org.wso2.carbon.ml.integration.common.utils.MLIntegrationTestConstants;
+import org.wso2.carbon.ml.integration.common.utils.exception.MLHttpClientException;
+import org.wso2.carbon.ml.integration.common.utils.exception.MLIntegrationBaseTestException;
+
+/**
+ * Contains test cases related to building models for learning algorithms
+ */
+@Test(groups = "buildModels", dependsOnGroups = "getModels")
+public class LearningAlgorithmsTestCase extends MLBaseTest {
+
+    private MLHttpClient mlHttpclient;
+    private static String analysisName;
+    private static String modelName;
+    private static int analysisId;
+    private static int modelId;
+    private CloseableHttpResponse response;
+
+    /**
+     * Sets the configuration of the model to be trained
+     *
+     * @param algorithmName     Name of the learning algorithm
+     * @param algorithmType     Type of the learning algorithm
+     * @param response          Response attribute
+     * @param trainDataFraction Fraction of data from the dataset to be trained with
+     * @param projectID         ID of the project
+     * @param datasetOffset     Additional information about the name
+     * @throws MLHttpClientException
+     */
+    private void setConfiguration(String algorithmName, String algorithmType, String response,
+                                  String trainDataFraction, int projectID, String datasetOffset) throws MLHttpClientException {
+        analysisName = "TestAnalysis_" + algorithmName + datasetOffset;
+
+        //Create an analysis
+        mlHttpclient.createAnalysis(analysisName, projectID);
+        analysisId = mlHttpclient.getAnalysisId(analysisName);
+
+        //Set Model Configurations
+        Map<String, String> configurations = new HashMap<String, String>();
+        configurations.put(MLIntegrationTestConstants.ALGORITHM_NAME, algorithmName);
+        configurations.put(MLIntegrationTestConstants.ALGORITHM_TYPE, algorithmType);
+        configurations.put(MLIntegrationTestConstants.RESPONSE, response);
+        configurations.put(MLIntegrationTestConstants.TRAIN_DATA_FRACTION_CONFIG, trainDataFraction);
+        mlHttpclient.setModelConfiguration(analysisId, configurations);
+
+        //Set default Hyper-parameters
+        mlHttpclient.doHttpPost("/api/analyses/" + analysisId + "/hyperParams/defaults", null);
+
+        // Create a model
+        CloseableHttpResponse httpResponse = mlHttpclient.createModel(analysisId, org.wso2.carbon.ml.integration.common.utils.MLIntegrationTestConstants.VERSIONSET_ID);
```

Comment:
You might have to pass the VersionSetId of each of the dataset created, for each of the project, when creating the model.